middle_end/flambda2/simplify/simplify_set_of_closures.ml

```diff
@@ -820,8 +820,14 @@ let simplify_set_of_closures0 context set_of_closures ~closure_bound_names
         let closure_offsets =
           match uacc_after_upwards_traversal with
           | None -> closure_offsets
-          | Some uacc_after_upwards_traversal ->
-            UA.closure_offsets uacc_after_upwards_traversal
+          | Some uacc_after_upwards_traversal -> (
+            let closure_offsets =
+              DA.closure_offsets (UA.creation_dacc uacc_after_upwards_traversal)
+            in
+            match UA.closure_offsets uacc_after_upwards_traversal with
+            | Unknown -> closure_offsets
+            | Known offsets ->
+              Code_id.Map.add new_code_id offsets closure_offsets)
```

Comment:
I'm not certain this code is correct. The suspicious thing is that we effectively ignore the `closure_offsets` accumulated (i.e. the one we e.g. return at line 822 just above), which seems wrong.

Reply:
We're not ignoring them, we've passed them to `simplify_function` so they're already present in `uacc_after_upwards_traversal`. It's only when `simplify_function` doesn't actually do any simplification that `uacc_after_upwards_traversal` is `None`, in which case we return the old `closure_offsets` directly.
Note that `simplify_function` may have added its own entries to the closure offsets map (if it contains a direct partial application for example) so it's incorrect to use anything other than the `dacc` at the end of the function.

Reply:
This sounds like it needs a comment

Reply:
Ah yeah I've seen the context now whilst reviewing the other PR.